specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json

```diff
@@ -3811,6 +4647,164 @@
         }
       }
     },
+    "StaticSiteDatabaseConnectionConfigurationFileOverview": {
+      "description": "A database connection configuration file",
+      "type": "object",
+      "properties": {
+        "fileName": {
+          "description": "The name of the configuration file.",
+          "type": "string",
+          "readOnly": true
+        },
+        "contents": {
+          "description": "The Base64 encoding of the file contents.",
+          "type": "string",
+          "readOnly": true
+        },
+        "type": {
+          "description": "The type of configuration file.",
+          "type": "string",
+          "readOnly": true
+        }
+      },
+      "x-ms-client-flatten": true
+    },
+    "StaticSiteDatabaseConnectionOverviewARMResource": {
+      "description": "Static Site Database Connection Overview ARM resource.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "./CommonDefinitions.json#/definitions/ProxyOnlyResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "description": "StaticSiteDatabaseConnectionOverviewARMResource resource specific properties",
+          "type": "object",
+          "properties": {
+            "resourceId": {
+              "description": "The resource id of the database.",
+              "type": "string",
+              "readOnly": true
+            },
+            "connectionIdentity": {
+              "description": "If present, the identity is used in conjunction with ConnectionString to connect to the database.",
+              "type": "string",
+              "readOnly": true
+            },
+            "connectionString": {
+              "description": "The connection string to use to connect to the database.",
+              "type": "string",
+              "readOnly": true
+            },
+            "region": {
+              "description": "The region of the database resource.",
+              "type": "string",
+              "readOnly": true
+            },
+            "configurationFiles": {
+              "description": "A list of configuration files associated with this database connection.",
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/StaticSiteDatabaseConnectionConfigurationFileOverview"
+              },
+              "x-ms-identifiers": [],
```

Comment:
Don't files have some unique id (or tuple of ids) you can address them with... like a path?

Reply:
Can you help clarify what level of uniqueness this is supposed to be (I tried looking at the docs but couldn't find a clear answer)? Should it be unique within the `configurationFiles` array, or some sort of globally unique id across all customers? Like within a `configurationFiles` array, all the file names would be different, but these file names could be the same in the configurationFiles of another database connection.

Reply:
My impression is that, as long as its unique within this particular array, so that it can be used for key-value lookup, its OK... I'm not totally sure about this however.

[https://azure.github.io/autorest/extensions/#x-ms-identifiers](https://azure.github.io/autorest/extensions/#x-ms-identifiers)

Reply:
@weidongxu-microsoft do you know?

Reply:
I am not sure either. I had the impression that it is used by ARM or some of its subservice.

Same as Tim, I assume it to be unique for the item of the array. In this particular API, it maybe "fileName" (that one site won't have 2 configure override on same file?).